library/std/src/sys/windows/mutex.rs

```diff
@@ -37,106 +30,42 @@ pub type MovableMutex = Mutex;
 unsafe impl Send for Mutex {}
 unsafe impl Sync for Mutex {}
 
-struct Inner {
-    remutex: ReentrantMutex,
-    held: Cell<bool>,
-}
-
-#[derive(Clone, Copy)]
-enum Kind {
-    SRWLock,
-    CriticalSection,
-}
-
 #[inline]
 pub unsafe fn raw(m: &Mutex) -> c::PSRWLOCK {
-    debug_assert!(mem::size_of::<c::SRWLOCK>() <= mem::size_of_val(&m.lock));
-    &m.lock as *const _ as *mut _
+    m.srwlock.get()
 }
 
 impl Mutex {
     pub const fn new() -> Mutex {
         Mutex {
-            // This works because SRWLOCK_INIT is 0 (wrapped in a struct), so we are also properly
+            // This works because SRWLOCK_INIT is 0 (wrapped in a struct), so we are properly
             // initializing an SRWLOCK here.
```

Comment:
I'd just remove this comment here, as the code now uses `SRWLOCK_INIT` directly. So no need to explain it's equal to `0`.

Reply:
Done.